NDTensors/src/lib/Sectors/src/symmetry_style.jl

```diff
@@ -0,0 +1,17 @@
+# This file defines SymmetryStyle, a trait to distinguish abelian groups, non-abelian groups
+# and non-group fusion categories.
+
+using ..LabelledNumbers: LabelledInteger, label
+
+abstract type SymmetryStyle end
+
+struct AbelianStyle <: SymmetryStyle end
+struct NotAbelianStyle <: SymmetryStyle end
+
+combine_styles(::AbelianStyle, ::AbelianStyle) = AbelianStyle()
+combine_styles(::SymmetryStyle, ::SymmetryStyle) = NotAbelianStyle()
+
+SymmetryStyle(l::LabelledInteger) = SymmetryStyle(label(l))
+
+# crash for empty g. Currently impossible to construct.
+SymmetryStyle(g::AbstractUnitRange) = SymmetryStyle(first(g))
```

Comment:
The only way I could think of making the empty case work is basing it on the element type:
```julia
SymmetryStyle(a::AbstractUnitRange) = SymmetryStyle(eltype(a))
```
but then `SymmetryStyle` would have to work on types, which would require changing definitions on categories to this style:
```julia
SymmetryStyle(::Type{<:U1}) = AbelianStyle()
```
I think that would be reasonable, conceptually the style should just be based on the type and not the instance (i.e. determining the style shouldn't require runtime information like the sector label). Also there is precedence in the Julia language for basing traits on types rather than instances: https://docs.julialang.org/en/v1/base/arrays/#Base.Broadcast.BroadcastStyle.